### PR TITLE
Convert to Poetry

### PR DESCRIPTION
**Pull request by willsawyerrrr**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,44 +1 @@
...
-rich==13.7.0
-SecretStorage==3.3.3
-tomli==2.0.1
-tomlkit==0.12.3
-twine==4.0.2
-typing_extensions==4.9.0
-urllib3==2.1.0
-wrapt==1.16.0
-zipp==3.17.0
+# required by the `pylint` status check
```
Comment (andre-at-situ) [inline requirements.txt:1]: I'm not sure this is true anymore? I thought you updated the deploy pipe to fix this?

File: pyproject.toml [hunk 2]
```diff
@@ -1,28 +1,41 @@
...
 ]
-dynamic = ["version"]
 
-[project.urls]
-"Homepage" = "https://github.com/SituDevelopment/python3-entegy-API-wrapper"
-"Releases" = "https://github.com/SituDevelopment/python3-entegy-API-wrapper/releases"
+[tool.poetry.dependencies]
+python = "~=3.10.0"
+
+requests = "~=2.31.0"
```
Comment (andre-at-situ) [inline pyproject.toml:31]: It's nice to have a header on this section. Maybe use a comment?

File: requirements.txt (hunk 1 above)
Comment (andre-at-situ) [inline requirements.txt:1]: I don't think this is required anymore?

File: pyproject.toml (hunk 2 above)
Comment (willsawyerrrr) [reply pyproject.toml:31]: See https://github.com/SituDevelopment/integro-api/pull/244#discussion_r1472181551.

File: requirements.txt (hunk 1 above)
Comment (willsawyerrrr) [reply requirements.txt:1]: See https://github.com/SituDevelopment/fastapi-starter/pull/47#discussion_r1472170456.

Comment (willsawyerrrr) [reply requirements.txt:1]: See https://github.com/SituDevelopment/fastapi-starter/pull/47#discussion_r1472170456.

**Review by andre-at-situ — APPROVED** [review]

**Pull request merged**